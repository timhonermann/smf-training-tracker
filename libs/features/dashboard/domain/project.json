{
  "name": "dashboard-domain",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/features/dashboard/domain/src",
  "prefix": "stt",
  "projectType": "library",
  "tags": ["domain:dashboard", "type:domain"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
