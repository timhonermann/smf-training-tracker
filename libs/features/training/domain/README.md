# training-domain

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test training-domain` to execute the unit tests.
