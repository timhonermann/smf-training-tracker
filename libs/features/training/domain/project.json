{
  "name": "training-domain",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/features/training/domain/src",
  "prefix": "stt",
  "projectType": "library",
  "tags": [
    "domain:training",
    "type:domain"
  ],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
