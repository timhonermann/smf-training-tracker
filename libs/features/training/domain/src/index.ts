export * from './lib/state/training-location-store';
export * from './lib/state/training-store';
