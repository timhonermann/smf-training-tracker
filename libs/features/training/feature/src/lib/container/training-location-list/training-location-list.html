<stt-back-button [link]="backLink()" />

<div class="list-container">
  @for (location of store.entities(); track location.id) {
    <div class="list-item">{{ location.name }}</div>
  }
</div>

<stt-button (clicked)="store.navigateToTrainingLocationCreation()"
  >Create Training Location</stt-button
>
