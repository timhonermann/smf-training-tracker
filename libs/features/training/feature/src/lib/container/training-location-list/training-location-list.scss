:host {
  display: flex;
  height: 100%;
  flex-direction: column;
  gap: 16px;
}

.list-container {
  flex: 1;
  overflow: auto;
}

.list-item {
  height: 24px;
}
