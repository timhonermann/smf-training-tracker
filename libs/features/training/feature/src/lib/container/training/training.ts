import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'stt-training',
  imports: [],
  templateUrl: './training.html',
  styleUrl: './training.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class Training {}
