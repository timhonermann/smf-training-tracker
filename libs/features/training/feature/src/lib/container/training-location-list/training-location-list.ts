import {
  ChangeDetectionStrategy,
  Component,
  inject,
  signal,
} from '@angular/core';
import { BackButton, Button } from '@stt/shared/button/ui';
import { TrainingLocationStore } from '@stt/features/training/domain';
import { featureRoutes } from '@stt/shared/routing/model';

@Component({
  selector: 'stt-training-location-list',
  imports: [Button, BackButton],
  templateUrl: './training-location-list.html',
  styleUrl: './training-location-list.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TrainingLocationList {
  readonly #store = inject(TrainingLocationStore);

  readonly locations = this.#store.entities;

  readonly backLink = signal(`/${featureRoutes.TRAINING}`).asReadonly();

  navigateToTrainingLocationCreation(): void {
    this.#store.navigateToTrainingLocationCreation();
  }
}
