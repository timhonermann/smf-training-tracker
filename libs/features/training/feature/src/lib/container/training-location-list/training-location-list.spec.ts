import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TrainingLocationList } from './training-location-list';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { expect } from 'vitest';
import { TrainingLocationStore } from '@stt/features/training/domain';
import { MockProvider } from 'ng-mocks';

describe('TrainingLocationList', () => {
  let component: TrainingLocationList;
  let fixture: ComponentFixture<TrainingLocationList>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TrainingLocationList],
      providers: [
        MockProvider(TrainingLocationStore, {
          entities: vi.fn(),
        }),
      ],
    })
      .overrideComponent(TrainingLocationList, {
        set: { imports: [], schemas: [CUSTOM_ELEMENTS_SCHEMA] },
      })
      .compileComponents();

    fixture = TestBed.createComponent(TrainingLocationList);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('backLink', () => {
    expect(component.backLink()).toMatchSnapshot();
  });
});
