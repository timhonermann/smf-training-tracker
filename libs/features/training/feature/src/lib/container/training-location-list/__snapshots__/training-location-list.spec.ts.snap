// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrainingLocationList > backLink 1`] = `"/training"`;
